.github/workflows/Func_Env_Build_Test_CI.yml

```diff
@@ -14,19 +14,85 @@ concurrency:
   cancel-in-progress: false
   
 jobs:
-  test:
-    name: test
+
+  unittest:
+    # run all python versions, in other jobs run the latest python
+    name: unittest
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [ '3.6', '3.7', '3.8', '3.9', '3.10' ]
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install flake8 pytest pytest-cov
+          if [[ -f requirements.txt ]]; then pip install -r requirements.txt; fi
+          if [[ -f tests_requirements.txt ]]; then pip install -r tests_requirements.txt; fi
+      - name: Lint with flake8
+        run: |
+          # stop the build if there are Python syntax errors or undefined names
+          flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
+          # exit-zero treats all errors as warnings. The GitHub editor is 127 chars wide
+          flake8 . --count --exit-zero --max-complexity=10 --max-line-length=127 --statistics
+      - name: ⚙️ Set Kubeconfig and hosts
+        env:
+          KUBECONFIG: ${{ secrets.FUNC_KUBECONFIG }}
+          RUNNER_PATH: ${{ secrets.RUNNER_PATH }}
+          OCP_HOSTS: ${{ secrets.FUNC_OCP_HOSTS }}
+        run: |
+          mkdir -p "$RUNNER_PATH/.kube/"
+          echo "$KUBECONFIG" > "$RUNNER_PATH/.kube/config"
```

Comment:
```cp "$KUBECONFIG" "$RUNNER_PATH/.kube/config"```

Reply:
"$KUBECONFIG" is not a file, it a string

Reply:
In that case, rename KUBECONFIG to something else, such as KUBECONFIG_CONTENTS.  KUBECONFIG normally refers to a filename, not the contents of the file.